constraints/system.air

```diff
@@ -14,43 +14,260 @@
 # │    1    │ fmp - Free memory pointer (procedure local memory region)            │
 # │    2    │ ctx - Current execution context                                      │
 # │    3    │ in_syscall - System call execution flag                              │
-# │   4-7   │ Function hash - Current/parent function digest (4 elements)          │
+# │   4-7   │ fn_hash - Current function digest (4 elements)                       │
 # └─────────┴──────────────────────────────────────────────────────────────────────┘
 #
+# PROCEDURE EXECUTION MODES:
 #
-# STATUS: Not fully implemented
+# The Miden VM supports five different ways to execute procedures, each with distinct
+# semantics for context management, memory isolation, and caller identification:
+#
+# ┌──────────┬─────────┬─────────┬──────────────┬─────────────┬──────────────────────┐
+# │ Operation│   ctx'  │   fmp'  │   fn_hash'   │ in_syscall' │      Purpose         │
+# ├──────────┼─────────┼─────────┼──────────────┼─────────────┼──────────────────────┤
+# │ EXEC     │   ctx   │   fmp   │   fn_hash    │ in_syscall  │ Inline execution     │
+# │ CALL     │ clk + 1 │  2^30   │ Load new     │      0      │ Call known procedure │
+# │ SYSCALL  │    0    │  2^31   │ **PRESERVE** │      1      │ Call kernel          │
+# │ DYN      │   ctx   │   fmp   │ **PRESERVE** │ in_syscall  │ Dynamic inline exec  │
+# │ DYNCALL  │ clk + 1 │  2^30   │ Load new     │      0      │ Dynamic call         │
+# └──────────┴─────────┴─────────┴──────────────┴─────────────┴──────────────────────┘
+#
+# DETAILED SEMANTICS:
+#
+# 1. EXEC (no opcode, handled by MASM assembler):
+#    - Inline execution: procedure code is inserted directly into caller
+#    - No context switch, no register changes
+#    - Behaves like a macro expansion at compile time
+#    - Not enforced by AIR constraints (happens during assembly)
```

Comment:
As mentioned above, would remove from these docs

Reply:
I think it is still useful in the understanding of the rest, but have made it explicit that it is just an assembler-level concept